### PR TITLE
Feature/np 860 clean index

### PR DESCRIPTION
**Pull request by Carmendelope**
#### What does this PR do?
This PR implements a loop that deletes those indexes created X days ago (7 for now)
#### What are the relevant tickets?

- [NP-860](https://nalej.atlassian.net/browse/NP-860)

### CODE REVIEW

File: internal/app/slave/expire/expire.go [hunk 1]
```diff
@@ -59,3 +71,59 @@ func (m *Manager) Expire(ctx context.Context, request *grpc.ExpirationRequest) (
...
+	log.Debug().Interface("indexList", indexList).Msg("listing the indexes")
+
+	for _, index := range indexList {
+		remove, err := m.checkRemoveIndex(index)
+		if err != nil {
+			log.Debug().Str("index", index).Msg("error checking the index")
+		} else {
+			log.Debug().Str("index", index).Bool("remove", remove).Msg("checking the index")
+			if remove {
+				err = m.Provider.RemoveIndex(context.Background(), index)
```
Comment (dhiguero) [inline internal/app/slave/expire/expire.go]: avoid `context.Background()`

File: internal/app/slave/service.go [hunk 2]
```diff
@@ -70,6 +70,8 @@ func (s *Service) Run() derrors.Error {
 	expireManager := expire.NewManager(elasticProvider)
 	handler := handler.NewHandler(searchManager, expireManager)
 
+	go expireManager.DeleteIndexLoop()
```
Comment (dhiguero) [inline internal/app/slave/service.go]: I think we discussed the possibility of having a flag to switch on/off this feature with on the default value. Should we added it?

File: pkg/provider/loggingstorage/elasticsearch.go [hunk 3]
```diff
@@ -133,3 +137,44 @@ func (es *ElasticSearch) Expire(ctx context.Context, request *entities.SearchReq
...
+	client, dErr := es.Connect()
+	if dErr != nil {
+		return dErr
+	}
+	exists, err := client.IndexExists(index).Do(ctx)
+	if err != nil {
+		return derrors.NewInternalError("elastic ask for an index query failed", err)
+	}
+	if exists {
+		_, err := client.DeleteIndex(index).Do(context.Background())
```
Comment (dhiguero) [inline pkg/provider/loggingstorage/elasticsearch.go]: avoid `context.Background()`

Comment (Carmendelope) [reply pkg/provider/loggingstorage/elasticsearch.go]: Done

File: internal/app/slave/expire/expire.go [hunk 4]
```diff
@@ -59,3 +71,59 @@ func (m *Manager) Expire(ctx context.Context, request *grpc.ExpirationRequest) (
...
+		return true, nil
+	}
+	return false, nil
+}
+
+// deleteIndex gets all the indexes and removes the old ones
+func (m *Manager) deleteIndex() {
+	log.Debug().Msg("Delete Index")
+
+	indexList, err := m.Provider.GetIndexList(context.Background())
```
[the remark this answers is not in the document]
Comment (Carmendelope) [reply internal/app/slave/expire/expire.go]: Done

File: internal/app/slave/service.go (hunk 2 above)
Comment (Carmendelope) [reply internal/app/slave/service.go]: --expireLogs flag included

File: internal/app/slave/expire/expire.go [hunk 5]
```diff
@@ -30,6 +33,15 @@ import (
 	grpc "github.com/nalej/grpc-unified-logging-go"
 )
 
+// expire logs timer
+
+const LoopSleep = time.Minute * 60 * 24
+
+const defaultDaysExpired = 7
```
[the remark this answers is not in the document]
Comment (Carmendelope) [reply internal/app/slave/expire/expire.go]: Perfect. Changed to `DefaultLogEntryTTL`

File: internal/app/slave/expire/expire.go [hunk 6]
```diff
@@ -30,6 +33,15 @@ import (
...
+
+// TODO: the index name is associated to filebeat version. It must be taken into account when changing the session
+const indexPrefix = "filebeat-6.6.0-"
```
[the remark this answers is not in the document]
Comment (Carmendelope) [reply internal/app/slave/expire/expire.go]: Done

**Comment by Carmendelope** [comment]
Requested changes done

File: internal/app/slave/expire/expire.go [hunk 7]
```diff
@@ -59,3 +72,69 @@ func (m *Manager) Expire(ctx context.Context, request *grpc.ExpirationRequest) (
...
+	log.Debug().Msg("Delete Index")
+
+	listCtx, listCancel := utils.GetContext()
+	defer listCancel()
+	indexList, err := m.Provider.GetIndexList(listCtx)
+	if err != nil {
+		log.Warn().Str("err", err.DebugReport()).Msg("error cleaning index")
+		return
+	}
+	log.Debug().Interface("indexList", indexList).Msg("listing the indexes")
```
Comment (dhiguero) [inline internal/app/slave/expire/expire.go]:
```suggestion
```

File: internal/app/slave/expire/expire.go [hunk 8]
```diff
@@ -59,3 +72,69 @@ func (m *Manager) Expire(ctx context.Context, request *grpc.ExpirationRequest) (
...
+
+	for _, index := range indexList {
+		remove, err := m.checkRemoveIndex(index)
+		if err != nil {
+			log.Debug().Str("index", index).Msg("error checking the index")
```
Comment (dhiguero) [inline internal/app/slave/expire/expire.go]:
```suggestion
			log.Warn().Str("index", index).Msg("error checking the index")
```

File: internal/app/slave/expire/expire.go [hunk 9]
```diff
@@ -59,3 +72,69 @@ func (m *Manager) Expire(ctx context.Context, request *grpc.ExpirationRequest) (
...
+		} else {
+			log.Debug().Str("index", index).Bool("remove", remove).Msg("checking the index")
+			if remove {
+				ctx, cancel := utils.GetContext()
+				err = m.Provider.RemoveIndex(ctx, index)
```
Comment (dhiguero) [inline internal/app/slave/expire/expire.go]:
```suggestion
defer cancel()
				err = m.Provider.RemoveIndex(ctx, index)
```

File: internal/app/slave/expire/expire.go [hunk 10]
```diff
@@ -59,3 +72,69 @@ func (m *Manager) Expire(ctx context.Context, request *grpc.ExpirationRequest) (
...
+				if err != nil {
+					log.Warn().Str("index", index).Str("err", err.DebugReport()).Msg("error cleaning index")
+				}
+				cancel()
```
Comment (dhiguero) [inline internal/app/slave/expire/expire.go]:
```suggestion
```

File: internal/app/slave/expire/expire.go (hunk 7 above)
Comment (Carmendelope) [reply internal/app/slave/expire/expire.go]: removed

File: internal/app/slave/expire/expire.go (hunk 8 above)
Comment (Carmendelope) [reply internal/app/slave/expire/expire.go]: changed

**Pull request merged**